static/js/reducers/ui_test.js

```diff
@@ -0,0 +1,100 @@
+import {
+  CLEAR_UI,
+  UPDATE_DIALOG_TEXT,
+  UPDATE_DIALOG_TITLE,
+  SET_DIALOG_VISIBILITY,
+  SET_WORK_HISTORY_EDIT,
+  SET_WORK_DIALOG_VISIBILITY,
+  SET_WORK_DIALOG_INDEX,
+
+  clearUI,
+  updateDialogText,
+  updateDialogTitle,
+  setDialogVisibility,
+  setWorkHistoryEdit,
+  setWorkDialogVisibility,
+  setWorkDialogIndex,
+} from '../actions/ui';
+
+import configureTestStore from 'redux-asserts';
+import rootReducer from '../reducers';
+import assert from 'assert';
+import sinon from 'sinon';
+
+describe('ui reducers', () => {
+  let sandbox, store, dispatchThen;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    store = configureTestStore(rootReducer);
+    dispatchThen = store.createDispatchThen(state => state.ui);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+
+    store = null;
+    dispatchThen = null;
+  });
+
+  it('should set a dialog title', done => {
+    dispatchThen(updateDialogTitle('A title'), [UPDATE_DIALOG_TITLE]).then(state => {
+      assert.equal(state.dialog.title, 'A title');
+      done();
+    });
+  });
+
+  it('should set dialog text', done => {
+    dispatchThen(updateDialogText('Some Text'), [UPDATE_DIALOG_TEXT]).then(state => {
+      assert.equal(state.dialog.text, 'Some Text');
+      done();
+    });
+  });
+
+  it('should set dialog visibility', done => {
+    dispatchThen(setDialogVisibility(true), [SET_DIALOG_VISIBILITY]).then(state => {
+      assert.equal(state.dialog.visible, true);
+      done();
+    });
+  });
+
+  it('should clear the ui', done => {
+    dispatchThen(clearUI(), [CLEAR_UI]).then(state => {
+      assert.deepEqual(state, {});
+      done();
+    });
+  });
+
+  it('should set the work history dialog visibility', done => {
+    dispatchThen(setWorkDialogVisibility(true), [SET_WORK_DIALOG_VISIBILITY]).then(state => {
+      assert.equal(state.workDialogVisibility, true);
+    });
+
+    dispatchThen(setWorkDialogVisibility(false), [SET_WORK_DIALOG_VISIBILITY]).then(state => {
```

Comment:
I think you'll need to move this within the `then` above, or else you may reach `done` without executing the `then` above